packages/react/src/components/Modal/Modal.types.ts

```diff
@@ -124,6 +124,14 @@ export interface IModalProps extends React.RefAttributes<HTMLDivElement>, IAcces
    */
   isModeless?: boolean;
 
+  /**
+   * Determines whether the dialog will be a Dialog or AlertDialog
+   * If this is true, then isBlocking and isModeless are ignored
```

Comment:
This should probably be updated to state more generally that "If this is set, it overrides the ARIA role determined by isBlocking and isModeless."

Both because it should override the role whether it's set to true or false, and also because isBlocking and isModeless won't entirely be ignored -- they'll still control both blocking and modal/modeless behavior :)